Project knight movement

Theo-Stefan opened the pull request:
I'm pleased to say that I've finally completed my task.
So basically I wanted to add Projections that show everytime where the knight is able to move.
I have simply added a switch that turns ON and OFF depending whether the user wants that or not, and when its ON  there are those small green circles on the board indicating that the knight can move there.

I want to note that this is my first ever contribution to a github repository and first time I write in javascript so the code might not be as laconic as it should, but I assure you that it does the work!

However I'd love to see your review on it and suggest me any improvements if there are!


![image](https://user-images.githubusercontent.com/72577715/214613851-3ad05ac7-8feb-48cc-9d39-9225f9e16764.png)

acenturyandabit commented:
Hi Theo-Stefan! 

Great job, I pulled it to my computer and it works! The button you chose looks very professional :)

As you mentioned, the code does have a lot of repetition. For instance I see you've got 8 different `<img>` elements, one for each blob. You could generate those using javascript (`document.createElement`) rather than putting them in the HTML manually, and then refer to them as indexes in an array, to make it easier to iterate through. But we can leave that for some other dev to clean up as a refactoring task.

One tiny thing to fix right now however: Please add "pointer-events: none" to your CSS for the <img>. This ensures that when the green dots are clicked on, the event will fall through to the tiles. Right now, clicking the green dot will not cause the knight to move. If you fix this, then I will approve :)

Thanks for your contribution!

Steven

Theo-Stefan commented:
Hey Steven, thanks for reviewing my pull request!
Also thanks for telling me about the "pointer-events: none" command because i really was looking for something like that and couldn't find it, so I was torturing myself trying to find other ways to solve the issue!! For example, I tried to set the z-index to -2.
So I added that command you suggested and deleted some unecessary code, the dark_mode.js looks a lot better now.

And yes there is a lot of unecessary repetition, hopefully some more experienced dev than me can take care of that :)
You can check my latest changes and let me know how it is, I hope it's alright!

acenturyandabit commented:
Looks great, accepted!
